### PR TITLE
{bio}[foss/2023a] EVidenceModeler v2.1.0, FASTA v36.3.8i, PASA v2.5.3, pblat v2.5.1

### PR DESCRIPTION
**Pull request by laraPPr**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by laraPPr** [comment]
I had not tested the sanity check command for EVidenceModeler yet. The one that is now included fails with this:
```
Sanity check failed: sanity check command $EVM_HOME/EVidenceModeler -h exited 

with code 255 (output: 

################# Evidence Modeler ##############################

#

#  parameters:

#

#  Required:

# 

#  --sample_id <str>           sample_id (for naming outputs)

#  --genome <str>              geno (took 36 secs)
```
The output looks correct, its just the exit code where EasyBuild fails over.

Shall I remove `sanity_check_commands` or is their a work around for this @boegel?

**Comment by laraPPr** [comment]
Test report by @laraPPr
**FAILED**
Build succeeded for 4 out of 5 (5 easyconfigs in total)
node4003.donphan.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 545.23.08, Python 3.11.3
See https://gist.github.com/laraPPr/761ee2d50d4b491c40bcf163527a11ce for a full test report.

**Comment by boegel** [comment]
@laraPPr I would use `"$EVM_HOME/EVidenceModeler -h | grep 'Evidence Modeler' "` as sanity check command.

You may need to also redirect stderr to stdout, so: `"$EVM_HOME/EVidenceModeler -h 2>&1 | grep 'Evidence Modeler' "`

**Comment by laraPPr** [comment]
Test report by @laraPPr
**SUCCESS**
Build succeeded for 0 out of 0 (5 easyconfigs in total)
node4002.donphan.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 545.23.08, Python 3.11.3
See https://gist.github.com/laraPPr/a73c462cd71cf1f59a235f57ca56bd6e for a full test report.

**Comment by laraPPr** [comment]
@boegelbot please test @ generoso

**Comment by laraPPr** [comment]
@boegelbot please test @ jsc-zen3

**Comment by laraPPr** [comment]
@boegelbot please test @ jsc-zen3

**Comment by laraPPr** [comment]
@boegelbot please test @ jsc-zen3

**Comment by laraPPr** [comment]
@boegelbot please test @ generoso

**Comment by laraPPr** [comment]
@boegel I  don't understand why GMAP-GSNAP is failing on Generoso?

**Comment by boegel** [comment]
@laraPPr There's an `#ifndef __haswell__` check in `src/select64-common.h`, so a different code path is used on Haswell (like `generoso`) and non-Haswell systems...

In the part for Haswell systems, there's this line:
```c
uint64_t result = uint64_t(1) << k;
```

which is invalid C, it should be:
```
uint64_t result = (uint64_t)1 << k;
```

So, a patch should work:
```diff
--- src/select64-common.h.orig  2024-10-15 13:51:30.954083781 +0000
+++ src/select64-common.h       2024-10-15 13:52:58.596323984 +0000
@@ -67,7 +67,7 @@
        return place + kSelectInByte[((x >> place) & 0xFF) | (byteRank << 8)];
 #elif defined(__GNUC__) || defined(__clang__)
        // GCC and Clang won't inline the intrinsics.
-       uint64_t result = uint64_t(1) << k;
+       uint64_t result = (uint64_t)1 << k;

        asm("pdep %1, %0, %0\n\t"
                "tzcnt %0, %0"
```

**Comment by laraPPr** [comment]
@boegelbot please test @ generoso

**Comment by laraPPr** [comment]
Test report by @laraPPr
**SUCCESS**
Build succeeded for 5 out of 5 (4 easyconfigs in total)
node4010.donphan.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 545.23.08, Python 3.11.3
See https://gist.github.com/laraPPr/21002c247e8de42c525ad8f16ab56a73 for a full test report.

**Comment by laraPPr** [comment]
@boegelbot please test @ jsc-zen3

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
node3124.skitty.os - Linux RHEL 9.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.9.18
See https://gist.github.com/boegel/3a4abda5d00f2b1ab97acd06fb3193d1 for a full test report.

File: easybuild/easyconfigs/e/EVidenceModeler/EVidenceModeler-2.1.0-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+}]
+checksums = [None]
+
+dependencies = [
+    ('PASA', '2.5.3',),
+]
+
+# Install ParaFly
+postinstallcmds = ["cd %(installdir)s && make"]
```
[the remark this answers is not in the document]
Comment (laraPPr) [reply easybuild/easyconfigs/e/EVidenceModeler/EVidenceModeler-2.1.0-foss-2023a.eb:28]: it fails with this when I try what you suggested `configure: error: C++ compiler cannot create executables`

[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyconfigs/e/EVidenceModeler/EVidenceModeler-2.1.0-foss-2023a.eb:28]: We should also remove the pre-built binary:
```suggestion
postinstallcmds = ["cd %(installdir)s && rm plugin/ParaFly/bin/ParaFly && make"]
```

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/e/EVidenceModeler/EVidenceModeler-2.1.0-foss-2023a.eb (hunk 1 above)
[the remark this answers is not in the document]
Comment (laraPPr) [reply easybuild/easyconfigs/e/EVidenceModeler/EVidenceModeler-2.1.0-foss-2023a.eb:28]: EVidenceModeler is no longer maintained so I report it upstream but it probably is not gonna be picked up

**Comment by laraPPr** [comment]
Test report by @laraPPr
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in total)
node4012.donphan.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 545.23.08, Python 3.11.3
See https://gist.github.com/laraPPr/138139ec2aaba78487232851e2982d44 for a full test report.

**Comment by laraPPr** [comment]
Test report by @laraPPr
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
node4012.donphan.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 545.23.08, Python 3.11.3
See https://gist.github.com/laraPPr/283f99cbe8e42c9aaf34dd18262a61e5 for a full test report.

File: easybuild/easyconfigs/e/EVidenceModeler/EVidenceModeler-2.1.0-foss-2023a.eb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+sanity_check_paths = {
+    'files': ['EVidenceModeler', 'plugins/ParaFly/bin/ParaFly'],
+    'dirs': ['plugins/ParaFly', 'testing'],
+}
+
+modextrapaths = {
+    'EVM_HOME': '',
+}
+
+sanity_check_commands = ["$EVM_HOME/EVidenceModeler -h 2>&1 | grep 'Evidence Modeler'"]
```
Comment (boegel) [inline easybuild/easyconfigs/e/EVidenceModeler/EVidenceModeler-2.1.0-foss-2023a.eb:40]: @laraPPr Shouldn't we add the top-level installation directory to `$PATH`?
```suggestion
modextrapaths = {
    'EVM_HOME': '',
    'PATH': ''.
}

sanity_check_commands = ["EVidenceModeler -h 2>&1 | grep 'Evidence Modeler'"]

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/e/EVidenceModeler/EVidenceModeler-2.1.0-foss-2023a.eb (hunk 2 above)
Comment (laraPPr) [reply easybuild/easyconfigs/e/EVidenceModeler/EVidenceModeler-2.1.0-foss-2023a.eb:40]: I'm not sure funannotate only uses $EVM_HOME and that is also what their own documentation says. But in a sense their is no harm in adding it. I think.

File: easybuild/easyconfigs/p/PASA/PASA-2.5.3-foss-2023a.eb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    'cdbyank',
+    'cln2qual',
+    'mdust',
+    'psx',
+    'seqclean',
+    'seqclean.psx',
+    'slclust',
+    'trimpoly',
+]
+
```
Comment (laraPPr) [inline easybuild/easyconfigs/p/PASA/PASA-2.5.3-foss-2023a.eb:45]: This env variable needs to be set
```suggestion
modextrapaths = {
    'PASAHOME': '',
}
```

**Comment by laraPPr** [comment]
Test report by @laraPPr
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
node4012.donphan.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 545.23.08, Python 3.11.3
See https://gist.github.com/laraPPr/aeb63bd183a002b35dcf33a31d6395a6 for a full test report.

**Comment by laraPPr** [comment]
@boegelbot please test @ generoso

**Comment by laraPPr** [comment]
@boegelbot please test @ jsc-zen3

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
node3115.skitty.os - Linux RHEL 9.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.9.18
See https://gist.github.com/boegel/c9be57e99a9c84fa1262a3ebd15ff3cf for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**